### ISSUE TITLE
Last Run not loading for DAGs on the listing page

### ISSUE THREAD

**Issue by gargrishabh75** [apache/airflow#18286]
### Apache Airflow version

2.1.3 (latest released)

### Operating System

Debian GNU/Linux 10 (buster)

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==2.1.0
apache-airflow-providers-celery==2.0.0
apache-airflow-providers-cncf-kubernetes==2.0.2
apache-airflow-providers-docker==2.1.0
apache-airflow-providers-elasticsearch==2.0.2
apache-airflow-providers-ftp==2.0.0
apache-airflow-providers-google==5.0.0
apache-airflow-providers-grpc==2.0.0
apache-airflow-providers-hashicorp==2.0.0
apache-airflow-providers-http==2.0.0
apache-airflow-providers-imap==2.0.0
apache-airflow-providers-microsoft-azure==3.1.0
apache-airflow-providers-mysql==2.1.0
apache-airflow-providers-postgres==2.0.0
apache-airflow-providers-redis==2.0.0
apache-airflow-providers-sendgrid==2.0.0
apache-airflow-providers-sftp==2.1.0
apache-airflow-providers-slack==4.0.0
apache-airflow-providers-sqlite==2.0.0
apache-airflow-providers-ssh==2.1.0

### Deployment

Other Docker-based deployment

### Deployment details

Base docker image- apache/airflow:2.1.3-python3.7

### What happened

Last Runs for the DAGs on the listing page are not loading because the `/last_dagruns` endpoint is returning 500 error.

### What you expected to happen

The Last Run for the DAGs should show the latest execution date of each DAG 

### How to reproduce

1. Open the DAG listing page
2. Look at the `Last Run` column, which would continuously show the loader for each DAG
3. If you open the Network, you would observe a 500 error for `/last_dagruns` endpoint

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
CAn you please take a look at webserver logs and paste the error you see there? I think that must be something specific to your setting (I have some guesses but they are pretty wild without logs).

**Comment by gargrishabh75** [comment]
> CAn you please take a look at webserver logs and paste the error you see there? I think that must be something specific to your setting (I have some guesses but they are pretty wild without logs).

```Python version: 3.7.11
Airflow version: 2.1.3
-------------------------------------------------------------------------------
Traceback (most recent call last):
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/flask/app.py&#34;, line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/flask/app.py&#34;, line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/flask/app.py&#34;, line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/flask/_compat.py&#34;, line 39, in reraise
    raise value
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/flask/app.py&#34;, line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/flask/app.py&#34;, line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/airflow/www/auth.py&#34;, line 34, in decorated
    return func(*args, **kwargs)
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py&#34;, line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/airflow/www/views.py&#34;, line 858, in last_dagruns
    for r in query
  File &#34;/home/airflow/.local/lib/python3.7/site-packages/airflow/www/views.py&#34;, line 858, in &lt;dictcomp&gt;
    for r in query
AttributeError: &#39;NoneType&#39; object has no attribute &#39;isoformat&#39;

**Comment by imamdigmi** [comment]
yes, I encounter the same problem
![image](https://user-images.githubusercontent.com/10998240/133712122-7aee68ee-dee5-4428-b2de-7b889b492aca.png)

**Comment by uranusjr** [comment]
This is duplicate to #17814 and fixed in main in #17732. Note that the PR is currently not backported to 2.1.
